### ISSUE TITLE
Android: touch events are executed twice when talkback is enabled

### ISSUE THREAD

**Issue by IvanIhnatsiuk** [software-mansion/react-native-gesture-handler#2808]
### Description

Hello, thanks for your library it is really good and very powerful 😎 .

I found an issue that when talkback is enabled, the onPress call occurs multiple times when interacting with the buttons. 

https://github.com/software-mansion/react-native-gesture-handler/assets/86000012/8c37625f-75b7-4a8f-bce5-d20e29171b99

In the video the goBack callback was called twice, so I was navigated to the Accounts tab :)

### Steps to reproduce

1. Enable talkback on android emulator
2. Run your example application on android emulator/real device
3. Press on any button
4. observe that onPress was executed twice

### Snack or a link to a repository

https://github.com/software-mansion/react-native-gesture-handler

### Gesture Handler version

2.15.0

### React Native version

0.72.10

### Platforms

Android

### JavaScript runtime

Hermes

### Workflow

React Native (without Expo)

### Architecture

Paper (Old Architecture)

### Build type

Debug mode

### Device

Android emulator

### Device model

Pixel 7 pro

### Acknowledgements

Yes

**Comment by m-bert** [comment]
Hi! I've just tested that on my Samsung S20+ and Pixel 6a emulator - both work fine. To test that, I've added logs into onPress callbacks in `RectButtons`. Could you provide a reproduction that we can look at?

**Comment by IvanIhnatsiuk** [comment]
@m-bert Hello,  I think you can reproduce this in your example app. To reproduce this, I simply updated the App.tsx file to 

```
import React from 'react';
import {
  BorderlessButton,
  GestureHandlerRootView,
} from 'react-native-gesture-handler';
import { Text } from 'react-native';

const press = () => console.log('pressed');

export default function App() {
  return (
    <GestureHandlerRootView style={{ justifyContent: 'center', flex: 1 }}>
      <BorderlessButton onPress={press}>
        <Text>Press me</Text>
      </BorderlessButton>
    </GestureHandlerRootView>
  );
}
```


https://github.com/software-mansion/react-native-gesture-handler/assets/86000012/e529c3f2-9861-4c69-a248-1300606cc831

Please, let me know if you need more details :)

**Comment by m-bert** [comment]
Hello @IvanIhnatsiuk! I've copied your code and everything is fine:

https://github.com/software-mansion/react-native-gesture-handler/assets/63123542/dabf65fe-29b7-41a2-a4c0-4b08580eccb5

It also works on my physical Samsung S20+

**Comment by m-bert** [comment]
I can see that you're using another emulator, I'll try to test that on Pixel 7.

**Comment by m-bert** [comment]
Okay, I've tested that on Pixel 7 Pro emulator and I still wasn't able to reproduce this issue.

**Comment by IvanIhnatsiuk** [comment]
@m-bert Sorry for the long reply, could you please try this code?

```
import React from 'react';
import {
  BorderlessButton,
  GestureHandlerRootView,
} from 'react-native-gesture-handler';
import { Text, View } from 'react-native';

const press = () => console.log('pressed');

export default function App() {
  return (
    <GestureHandlerRootView style={{ justifyContent: 'center', flex: 1 }}>
      <BorderlessButton onPress={press}>
        <View accessible accessibilityRole="button">
          <Text>Press me</Text>
        </View>
      </BorderlessButton>
    </GestureHandlerRootView>
  );
}

```

**Comment by m-bert** [comment]
Thank you @IvanIhnatsiuk! Now I can reproduce this issue. Seems like the problem lies in additional `View` inside `BorderlessButton` - without it it works fine, but I guess that iOS doesn't work in that case (I have yet to check that).

**Comment by IvanIhnatsiuk** [comment]
Yes, the problem is in additional view element. However, we need to add this view to make buttons accessible: https://docs.swmansion.com/react-native-gesture-handler/docs/components/buttons/

**Comment by IvanIhnatsiuk** [comment]
Also, Could you please remove "can't repro" label? 🙂

**Comment by m-bert** [comment]
>Also, Could you please remove "can't repro" label? 🙂

Sure, done 😄 

>However, we need to add this view to make buttons accessible

I'm aware of that. As far as I can tell buttons should work without this `View` on android. I'm not sure about iOS though since I don't have access to iPhone right now and testing accessibility on simulator is not pleasant. I'll get back to you when we find a solution to this problem 😅

**Comment by latekvo** [comment]
Hey @IvanIhnatsiuk, could you please let me know if PR #3002 has fixed this issue for you?
